server/src/test/java/org/candlepin/resource/HypervisorResourceTest.java

```diff
@@ -310,4 +312,82 @@ public void checkInCreatesNoNewConsumerWhenCreateIsFalse() throws Exception {
     public void ensureBadRequestWhenNoMappingIsIncludedInRequest() {
         hypervisorResource.hypervisorUpdate(null, principal, "an-owner", false);
     }
+
+    @Test
+    public void ensureEmptyHypervisorIdsAreIgnored() throws Exception {
+        Owner owner = new Owner("admin");
+
+        Map<String, ArrayList<GuestId>> hostGuestMap = new HashMap<String, ArrayList<GuestId>>();
+        hostGuestMap.put("", new ArrayList(Arrays.asList(new GuestId("GUEST_A"),
+                                         new GuestId("GUEST_B"))));
+        hostGuestMap.put("HYPERVISOR_A", new ArrayList(Arrays.asList(new GuestId("GUEST_C"),
+                                                     new GuestId("GUEST_D"))));
+        when(ownerCurator.lookupByKey(eq(owner.getKey()))).thenReturn(owner);
+
+        when(consumerCurator.getHostConsumersMap(any(Owner.class),
+             any(Set.class))).
+             thenReturn(new VirtConsumerMap());
+        when(consumerCurator.getGuestConsumersMap(any(Owner.class), any(Set.class))).
+            thenReturn(new VirtConsumerMap());
+
+        when(ownerCurator.lookupByKey(eq(owner.getKey()))).thenReturn(owner);
+        when(principal.canAccess(eq(owner), eq(SubResource.CONSUMERS), eq(Access.CREATE))).
+            thenReturn(true);
+        when(consumerTypeCurator.lookupByLabel(
+         eq(ConsumerTypeEnum.HYPERVISOR.getLabel()))).thenReturn(hypervisorType);
+        when(idCertService.generateIdentityCert(any(Consumer.class)))
+            .thenReturn(new IdentityCertificate());
+
+        HypervisorCheckInResult result = null;
+        try {
+            result = hypervisorResource.hypervisorUpdate(hostGuestMap,
+            principal, owner.getKey(), true);
+        }
+        catch (PersistenceException e) {
+            fail("Expected no exception");
```

Comment:
This isn't necessary.  The test method already throws and exception, and if an exception is thrown, JUnit marks the test as an "error" (which is actually what the result [should be](http://stackoverflow.com/questions/3425995/whats-the-difference-between-failure-and-error-in-junit) marked as)